crates/sui-adapter/src/execution_engine.rs

```diff
@@ -620,14 +620,21 @@ fn advance_epoch<S: BackingPackageStore + ParentSync + ChildObjectResolver>(
             new_package.compute_object_reference()
         );
 
+        let write_kind = if version == SequenceNumber::from(1) {
```

Comment:
nit: (not sure how realistic it is, but it would be good to avoid arbitrary conversions of u64 to version).

```suggestion
        let write_kind = if version == OBJECT_START_VERSION {
```